src/EPoetryClientFactory.php

```diff
@@ -107,6 +116,14 @@ public function getClient(): EPoetryClient
         $clientBuilder = new ClientBuilder($clientFactory, $this->wsdl, $this->options);
         $clientBuilder->withClassMaps(EPoetryClassmap::getCollection());
 
+        if ($this->logger) {
+            $logPlugin = new LoggerSubscriber($this->logger, $this->logLevel);
+            if (!$this->eventDispatcher) {
+                $this->eventDispatcher = new EventDispatcher();
+            }
```

Comment:
Given that we check again right here below if the event dispatcher is set shouldn't we do this in one place only right before the `if ($this->logger)`?

Reply:
I'm not sure how that would work: the logger needs to be added to the dispatcher before it is added in the client builder. If I check for the dispatcher and add it before checking for the logger, then I need to re-add the dispatcher if indeed there is a custom logger available.

Reply:
Something like this wouldn't work?

```php
        $this->eventDispatcher = $this->eventDispatcher ?? new EventDispatcher();
        $clientBuilder->withEventDispatcher($this->eventDispatcher);

        if ($this->logger) {
            $logger = new Logger($this->logger, $this->logLevel);
            $logPlugin = new LoggerSubscriber($logger);
            $this->eventDispatcher->addSubscriber($logPlugin);
        }

```